### PR TITLE
Blood runes now actually cost blood

### PR DESCRIPTION
**Pull request by ghost**
Title says it all

- Scribing blood runes now actually costs blood, a minute amount, but still more than the previous arbitrary 0.1-1 brute scribing used to inflict
- If you are a species without blood, you will need to improvise, either by getting blood on your hands in some way, or using pools of blood in your vicinity. Finally some reason to spread blood around besides aesthetic

Also standardises the message created from attempting to scribe to visible_message, rather than the abuse of show_message

:cl:
 * rscadd: Scribing blood runes now actually uses blood.
 * rscadd: Those species without blood will need to improvise with blood on the floor or on their hands.

### CODE REVIEW

**Comment by SonixApache** [comment]
all of my  ❤️ s

File: code/game/gamemodes/cult/ritual.dm [hunk 1]
```diff
@@ -463,10 +463,40 @@ var/global/list/rune_list = list() // HOLY FUCK WHY ARE WE LOOPING THROUGH THE W
...
+						for (var/obj/effect/decal/cleanable/blood/B in T)
+							if(B.amount && B.counts_as_blood)
+								user.visible_message("<span class='warning'>[user] paws at the blood puddles splattered on \the [T], and begins to chant and paint symbols on the floor.</span>",\
+								"<span class='warning'>You use the blood splattered across \the [T], and begin drawing a rune on the floor whilst chanting the ritual that binds your life essence with the dark arcane energies flowing through the surrounding world.</span>",\
+								"<span class='warning'>You hear chanting.</span>")
+								B.amount--
+								found = 1
+								break
+					if(!found)
+						to_chat(user, "<span class='notice'>You have no blood within, on you, or nearby, you can use to draw a rune!</span>")
```
Comment (N3X15) [inline code/game/gamemodes/cult/ritual.dm]: > or nearby, you

this needs rephrasing.

Comment (ghost) [reply code/game/gamemodes/cult/ritual.dm]: Fair point, hang on

Comment (Shadowmech88) [reply code/game/gamemodes/cult/ritual.dm]: `"You have no blood in, on, or around you that can be used draw a rune!"`

File: code/game/gamemodes/cult/ritual.dm [hunk 2]
```diff
@@ -463,10 +463,40 @@ var/global/list/rune_list = list() // HOLY FUCK WHY ARE WE LOOPING THROUGH THE W
...
+						for (var/obj/effect/decal/cleanable/blood/B in T)
+							if(B.amount && B.counts_as_blood)
+								user.visible_message("<span class='warning'>[user] paws at the blood puddles splattered on \the [T], and begins to chant and paint symbols on the floor.</span>",\
+								"<span class='warning'>You use the blood splattered across \the [T], and begin drawing a rune on the floor whilst chanting the ritual that binds your life essence with the dark arcane energies flowing through the surrounding world.</span>",\
+								"<span class='warning'>You hear chanting.</span>")
+								B.amount--
+								found = 1
+								break
+					if(!found)
+						to_chat(user, "<span class='notice'>You have no blood in, on, or around you can use to draw a rune!</span>")
```
Comment (N3X15) [inline code/game/gamemodes/cult/ritual.dm]: > around you can use

> around you that you can use

**Comment by Megamaw** [comment]
an excellent change. i look forward to seeing how this affects my play as both a skeleton and as a janitor.

**Comment by SonixApache** [comment]
i look forward to an age without skeletons and rigged skeleton votes

**Comment by Shadowmech88** [comment]
There was a change made some time ago so that any runes drawn by a player would be in the color of the player's own blood.
Make sure to make it so that if the player is drawing it via bloody hands/blood puddles, it uses the color of that blood instead of their own blood color.

**Comment by DamianX** [comment]
Nice, but this still keeps the brute damage, meaning you could powergame it through the CMC or something.

**Comment by Exxion** [comment]
I once had someone notice a tiny bruise on my hand or something near roundstart and conclude I was a cultist
They didn't out us, but dear fuck that was meta

**Comment by AI-Is-Rogue** [comment]
What about Dionae using their sap to draw runes? Because runes aren't the only place it shows up.

**Comment by ghost** [comment]
This is gonna make cult research damn near impossible to do as a bloodless species.  With this change, for each bloodless species starting cultist, you are further reducing the chances of it actually getting done.

**Comment by TheFriggenSawbones** [comment]
I mean whats the purpose of this change? It seems kinda pointless, just adding unnecessary stuff and disincentivizes the cult from converting certain races.

**Comment by Shadowmech88** [comment]
>damn near impossible

Just take a blood sample from someone, maybe a cult buddy, and splash it on the floor where you're going to do your research.

**Comment by ghost** [comment]
> Just take a blood sample from someone, maybe a cult buddy, and splash it on the floor where you're going to do your research.

If I am reading this correctly, it removes the blood from the surface/item/self for each rune you draw.  Meaning, you have to do that spilling process for each rune in the research process?  Golly, that sounds like a blast.

>maybe a cult buddy

Congratulations, you are now required to be babysat by a fellow cultist to do something that they could do alone.  At best, you are slowing them down with the logistics of the request, when they could just be doing it.

**Comment by Allspice** [comment]
Sounds like a good opportunity to remove the word research requirement

**Comment by DamianX** [comment]
It really doesn't, though. Feel free to open a discussion about it.

**Comment by ghost** [comment]
@HarseTheef it removes 1 amount from the used pool of blood

Each splash of blood has at least 3~5 'amount' to count as its depth, unless it's a 'drip' one like from a nosebleed or a blood track which arguably wouldn't be a very deep puddle.

Just roundhouse pun pun with a toolbox and get to work

@AI-is-rogue Dionae don't have a sap reagent to take from, they're bloodless beasts like skellingtons and plasmamen

**Comment by SonixApache** [comment]
fuck xenos

**Comment by nuklearcellphoneg** [comment]
I like it a lot, it seems like a really fun and intuitive update.

**Comment by yclatious** [comment]
Fuck xenos, bring in the blood

**Comment by ghost** [comment]
@Skullyton Neat, I was unaware of that.  Seems legit then.

While I am not a fan of tedium for the sake of tedium, it seems way more tolerable knowing that.

**Comment by AI-Is-Rogue** [comment]
@Skullyton 
They can splatter "sap" on the ground and (I think) on items.
I liked the idea of using that stuff for their runes.

**Comment by TheFriggenSawbones** [comment]
why does cult need a nerf

**Comment by Theophrastus1** [comment]
Doesn't seem like a good change at all, because you can just rush pun pun or rush monkey cubes and there you go, you now have a bunch of blood. Really fucking pointless.

It's just shifting around what the cult should prioritize and adding unnecessary steps to the prep work for cult. Also makes an already easy to be caught antag even easier to be caught. Just highly unnecessary and pointless since monkey cubes exist and bloodless species can get the blood necessary that way.

**Comment by yclatious** [comment]
Ive kinda come to agree with theo after he explained why he thinks this is a shit idea. As much as I want the ack cracked and shit, he did bring up some good points.

**Comment by ghost** [comment]
Then do so? I added alternative sources of blood you can use for a reason

**Comment by Theophrastus1** [comment]
Right, so then why even make this change when that's exactly what will go on, and nothing will have changed except bloodless species now have to go order a crate from cargo? It adds no difficulty, it just adds an extra step which is the worst kind of change next to reactionary.

**Comment by DamianX** [comment]
because my immersion
it's a blood cult

**Comment by Theophrastus1** [comment]
@DamianX Then make the act of drawing the rune make you bleed instead of just causing brute, no need for pointless changes like this that supposedly bring "difficulty" when it just changes how you do something in a worse manner.

**Comment by DamianX** [comment]
But that's what the PR does? You only need an external source of blood if you're a species without blood.

**Comment by ghost** [comment]
@Theophrastus1 If drawing a rune made you bleed, you'd be worse off than the act of drawing a rune taking some blood, as you'd need to keep patching yourself. Newbies would die just through experimenting, and bandages are a commodity not everyone can acquire. You'd be worse off.

**Comment by SonixApache** [comment]
@Theophrastus1 
Monkeys already get used for the spread blood objective, saying "oh but now they'll need monkeys" doesn't really carry any sway as argument.

And "My xeno rights!" is also really not a reason not to implement this.

Blood cultists needing to use blood to scribe blood runes that summon a blood god makes sense.

**Comment by yclatious** [comment]
Okay, hear me out

What if, when you are a ack or a plas-ack, if you don't have any blood puddles, you break a bone and write with the blood within? Cracks the Ack and makes shit work without monkeys

**Comment by SonixApache** [comment]
what if, when you are an ack or a plasmeme, you drag some poor blood-filled fucker into maint and use him to paint runes?

Or, hell, work together with your human cultist bretheren.

**Comment by DamianX** [comment]
I feel like this needs an "update wiki" tag.

**Comment by Megamaw** [comment]
don't add in some weird bullshit that lets skeletons bleed.

they're a bloodless race, and we should adjust our code as such.

is it a disadvantage? yes. but it's a small one, and it keeps it consistent with what's been established.

**Comment by TheFriggenSawbones** [comment]
@SonixApache or we can refrain from changing things for no real reason at all.

Reactionary balance prs, and their creators, are the worst.

**Comment by SonixApache** [comment]
"Writing blood runes as a blood cultist doesn't require blood" sounds like a bug that needs fixing to me, sawbones.

**Comment by TheFriggenSawbones** [comment]
thats a shitty excuse.

There was nothing wrong with it before and I doubt anyone actually asked for this change

**Comment by SonixApache** [comment]
I told you what's wrong with it. 
I asked for it plenty of times.

**Comment by Shadowmech88** [comment]
I believe I also requested it at some point.

File: code/game/gamemodes/cult/ritual.dm [hunk 3]
```diff
@@ -463,10 +463,40 @@ var/global/list/rune_list = list() // HOLY FUCK WHY ARE WE LOOPING THROUGH THE W
...
+			if(H.species.anatomy_flags & NO_BLOOD) //No blood, going to have to improvise
+				if(H.bloody_hands && H.held_items.len) //Blood on hand to use, and hands on hand to use
+					user.visible_message("<span class='warning'>[user] starts to paint drawings on the floor with the blood on their hands, whilst chanting.</span>",\
+					"<span class='warning'>You use the blood smeared on your hands to begin drawing a rune on the floor whilst chanting the ritual that binds your life essence with the dark arcane energies flowing through the surrounding world.</span>",\
+					"<span class='warning'>You hear chanting.</span>")
+					H.bloody_hands = max(0, H.bloody_hands - 1)
+				else //We'll have to search around for blood
+					var/turf/T = get_turf(user)
+					var/found = 0
+					if(H.held_items.len)
```
Comment (Intigracy) [inline code/game/gamemodes/cult/ritual.dm]: Why not just move all of this check to the top right under ishuman(user)?

Obviously if they don't have hands they can't draw.

File: code/game/gamemodes/cult/ritual.dm [hunk 4]
```diff
@@ -463,10 +463,40 @@ var/global/list/rune_list = list() // HOLY FUCK WHY ARE WE LOOPING THROUGH THE W
...
 		if(usr.get_active_hand() != src)
 			return
-		for (var/mob/V in viewers(src))
-			V.show_message("<span class='warning'>[user] slices open a finger and begins to chant and paint symbols on the floor.</span>", 1, "<span class='warning'>You hear chanting.</span>", 2)
-		to_chat(user, "<span class='warning'>You slice open one of your fingers and begin drawing a rune on the floor whilst chanting the ritual that binds your life essence with the dark arcane energies flowing through the surrounding world.</span>")
-		user.take_overall_damage((rand(9)+1)/10) // 0.1 to 1.0 damage
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			if(H.species.anatomy_flags & NO_BLOOD) //No blood, going to have to improvise
+				if(H.bloody_hands && H.held_items.len) //Blood on hand to use, and hands on hand to use
```
Comment (Intigracy) [inline code/game/gamemodes/cult/ritual.dm]: See other comment.

**Comment by Intigracy** [comment]
Also you should probably have a check for reagent containers that contain blood.

**Comment by ghost** [comment]
What, like they carry around a bucket of blood, or are you referring to the vessel container in people?

**Comment by DamianX** [comment]
I think he might be referring to species with the NO_BLOOD flag. If you inject those with blood it fills their reagent container instead of their vessel, IIRC.

**Comment by Intigracy** [comment]
I meant buckets/beakers

**Comment by ghost** [comment]
I still don't like this adding tedium and reducing the chances of research getting done in the game-mode infamous for its tedium required to the point of needing lessons to even be able to play it.

File: code/game/gamemodes/cult/ritual.dm (hunk 3 above)
Comment (Skullyton) [inline code/game/gamemodes/cult/ritual.dm]: bloody stumps? I dunno

**Comment by SonixApache** [comment]
@HarseTheef 
Is the issue the cult gamemode itself, the quality and volume of available information, or the players' willingness to learn?

**Comment by Theophrastus1** [comment]
I'd say it's the players' willingness to learn. A lot of people just still go out and kill, they aren't robust enough to keep kidnapping folks.

**Pull request merged**